### ISSUE TITLE
twisted.conch.test.test_cftp.TestOurServerCmdLineClient.testCommand fails when run as a user without a shell

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3914]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3914|
|Type|defect|
|Created|2009-07-14 22:54:37Z|
|Branch|https://github.com/twisted/twisted/tree/cftp-without-shell-3914|

This test runs `cftp` as a child process and then issues a shell escape command to it.  However, as the user running the process has no shell, the command cannot be evaluated.  The test should provide a fake _which user am I, and what is my shell_ implementation and exercise the usual case through this code path as well as the missing shell case (perhaps adding that can be split off into a separate ticket).

This test is failing on several builders which are now running on accounts without shells.

<details><summary>Searchable metadata</summary>

```
trac-id__3914 3914
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_cftp_without_shell_3914 branches/cftp-without-shell-3914
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1247612077000000 1247612077000000
changetime__1247838454000000 1247838454000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27143]) Branching to 'cftp-without-shell-3914'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

 * An annoying pyflake, not new here:
`twisted/conch/scripts/cftp.py:427: undefined name 'files'`

Please merge after a green build.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [27157]) Merge cftp-without-shell-3914

Author: exarkun
Reviewer: therve
Fixes: [#3914](https://github.com/twisted/twisted/issues/3914)

Add more tests for the possible behaviors of the shell escape feature of Conch's
sftp command line client, cftp.  In particular, test the case where the user has no
shell defined, and change the implementation to use /bin/sh in that case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
